### PR TITLE
red_rain game

### PR DESCRIPTION
**Pull request by matt-land**
This program adds a tiny arcade style game for the Microbit. It features a variety of game design elements and implements them in a simple way.

### CODE REVIEW

File: examples/red_rain.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    time = 130
+    score = 0
+    _streak = 0
+
+    def hit(self):
+        # register a hit
+        self._hits += 1
+        self._streak += 1
+        self.score += self._streak * self._level * 10
+        display.show(Image("99999:90009:90009:90009:99999"))
```
Comment (ntoll) [inline examples/red_rain.py]: why not `display.show(Image.SQUARE)`..?

File: examples/red_rain.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        sleep(25)
+
+    def miss(self):
+        # register a miss
+        self._misses += 1
+        self._streak = 0
+        display.show(Image("90009:09090:00900:09090:90009"))
```
Comment (ntoll) [inline examples/red_rain.py]: also, `display.show(Image.NO)` ;-)

Using built in images saves memory.

File: examples/red_rain.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        sleep(250)
+
+    def level_complete(self):
+        if self._hits < 10:
+            return
+        image = Image("00000:00900:09990:00900:00000")
```
Comment (ntoll) [inline examples/red_rain.py]: display.show(Image.YES) or something else..? List of all built-in images here: 

https://microbit-micropython.readthedocs.io/en/latest/image.html#attributes

File: examples/red_rain.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    field = [my_ball.draw_field(n) for n in range(4)]
+    field = ''.join(field) + paddle_sprites[paddle_position]
+    image = Image(field)
+    display.show(image)
+    my_score.level_complete()
+    if my_score.game_is_over():
+        break
+    sleep(my_score.time)
+
+display.scroll("G/O S {}".format(my_score.score), loop=True)
```
Comment (ntoll) [inline examples/red_rain.py]: G/O = Game over?

You can change the scroll speed with `delay=100` for example.

**Comment by matt-land** [comment]
I will try again with flattened branch

**Pull request closed without merging**